examples/adxl313_example.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2022 Analog Devices, Inc.
+#
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+#     - Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+import time
+
+import adi
+
+# Passs URI and device name as command line argument
+my_uri = sys.argv[1] if len(sys.argv) >= 2 else None
+my_dev = sys.argv[2] if len(sys.argv) >= 3 else None
+
+# Set up ADXL31x
+# Example: python3 adxl313_example.py serial:/dev/ttyACM0 ADXL313
+myacc = adi.adxl313(uri=my_uri, device_name=my_dev)
+# myacc = adi.adxl313(uri="serial:/dev/ttyACM0", device_name="ADXL313")
+# myacc = adi.adxl313(uri="ip:192.168.1.10", device_name="ADXL313")
+myacc.rx_buffer_size = 32
+myacc.rx_enabled_channels = [0, 1, 2]
+
+sfa = myacc.accel_x.sampling_frequency_available
+print("\nSample frequencies available:")
+print(sfa)
+
+print("\nInitial sample frequency:")
+print("Sample frequency: " + str(myacc.accel_x.sampling_frequency))
```

Comment:
Is the sampling frequency a per device or per channel attribute? If it's a per device attribute maybe it's worth treating it as such, but that may entail a change of the driver, so this is just something to think about.

Reply:
It is an channel attribute, that is shared between all channels. Once changed for one channel, it is valid for all the other ones. I used the same pattern as in other drivers and classes.